### PR TITLE
[Hinty] data.py / dadict.py / pton_ntop.py / mib.py

### PR DESCRIPTION
**Pull request by gpotter2**
- add typing to `data.py`, `dadict.py`, `pton_ntop.py`, `asn1/mib.py`
- Stop abusing `__dict__` of `DADict` to store data that simply shouldn't be there (`__dict__` are `Dict[str, Any]` and we used it wildly), and messes with mypy.
- ~~**requires https://github.com/secdev/scapy/pull/2864 (route & route6)**~~

### CODE REVIEW

File: scapy/dadict.py [hunk 1]
```diff
@@ -55,64 +73,84 @@ class DADict(object):
...
         """
         Return value that is used as key for the direct access
         """
-        return fixname(v)
+        if isinstance(v, (str, bytes)):
+            return fixname(v)
+        return "unknown"
 
     def update(self, *args, **kwargs):
+        # type: (*Dict[Any, Any], **Any) -> None
```
Comment (guedou) [inline scapy/dadict.py:91]: Could we use _K and _V here instead of Any?

File: scapy/data.py [hunk 2]
```diff
@@ -421,14 +446,15 @@ def reverse_lookup(self, name, case_sensitive=False):
         :returns: a dict of mac:tuples (Name, Extended Name)
         """
         if case_sensitive:
-            filtr = lambda x, l: any(x == z for z in l)
```
Comment (guedou) [inline scapy/data.py:424]: Why did you change `==` to `in`?

**Review by guedou — COMMENTED** [review]
Thanks for the PR. Did you manage to catch some bugs while typing these files?

File: scapy/data.py (hunk 2 above)
Comment (gpotter2) [reply scapy/data.py:424]: It's more convenient to match "Sun Microsystems" when just looking for "Sun".
This function already returned a dictionary of all MACs that match the OUI ordered by OUI, so it makes sense for the match to be wider.

**Comment by gpotter2** [comment]
See for instance https://github.com/secdev/scapy/issues/2896 which was catches with this PR.

File: scapy/data.py (hunk 2 above)
Comment (guedou) [reply scapy/data.py:424]: It makes. A separated PR will have make it easier to review =)

File: scapy/dadict.py (hunk 1 above)
Comment (gpotter2) [reply scapy/dadict.py]: Actually we can only use `_V`.
You can do `update(a=1)` -> `kwargs: Dict[str, _V]`
but not `update(1=2)`
so this function only works when `_K=str`...
